pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -1368,16 +1377,8 @@ func (gce *GCECloud) NodeAddresses(_ string) ([]api.NodeAddress, error) {
 	}, nil
 }
 
-func (gce *GCECloud) isCurrentInstance(instance string) bool {
-	return gce.instanceID == canonicalizeInstanceName(instance)
-}
-
 // ExternalID returns the cloud provider ID of the specified instance (deprecated).
 func (gce *GCECloud) ExternalID(instance string) (string, error) {
-	// if we are asking about the current instance, just go to metadata
```

Comment:
I'm worried about removing this extra check which was added by @cjcullen in https://github.com/kubernetes/kubernetes/pull/12028 to fix https://github.com/kubernetes/kubernetes/issues/12000. 

Also /cc @gmarek